### PR TITLE
Added max-xor-pair using a Trie.

### PR DESCRIPTION
**Pull request by rgkbitw**
Max xor pair occurring in an array can be solved using Trie.
This idea can be used in other problems as well.

[Resource](https://qr.ae/TWvjgq)

### CODE REVIEW

**Comment by Erfaniaa** [comment]
Please review this pull request. @poyea @siriak @AnupKumarPanwar

File: Graph/Dinic.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
+// Dinic's algorithm to find max flow in a graph.
```
Comment (poyea) [inline Graph/Dinic.cpp:1]: It seems they can be in two PRs next time.

File: Graph/Dinic.cpp [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+struct edge{
+	int from, to;
+	ll capacity, flow = 0;
```
Comment (poyea) [inline Graph/Dinic.cpp:4]: need a `using ll = long long`?

File: Graph/Dinic.cpp [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	edge(int from, int to, ll cap) : from(from) , to(to), capacity(cap) {}
+	void print(){
+		cout<<"("<<from<<"->"<<to<<") capacity:"<<capacity<<" Flow:"<<flow<<endl;
+	}
+};
+
+struct Dinic{
+	const ll INF = 1e18;
```
Comment (poyea) [inline Graph/Dinic.cpp:12]: `LONG_MAX`?

File: Others/maxXorPairusingTrie.cpp [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		for(int i=1; i<n; ++i){
+			int get = maxxor(arr[i]);
+			ans = max(get, ans);
+			insert(arr[i]);
+		}
+		return ans;
+    }
+};
+
+// Resource: https://qr.ae/TWvjgq
```
Comment (poyea) [inline Others/maxXorPairusingTrie.cpp:52]: move to the top

File: Graph/Dinic.cpp [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+			fill(ptr.begin(), ptr.end(), 0);
+			while(ll pushed= dfs(source, INF)){
+				f += pushed;
+			}
+		}
+		return f;
+	}
+};
+
+// see this soultion to understand how to use this: https://codeforces.com/problemset/submission/727/52666039
```
Comment (poyea) [inline Graph/Dinic.cpp:81]: move to the top

**Pull request closed without merging**